examples/seismic/model.py

```diff
@@ -642,13 +641,21 @@ def vp(self, vp):
         vp : float or array
             New velocity in km/s.
         """
-        self._vp = vp
-
         # Update the square slowness according to new value
         if isinstance(vp, np.ndarray):
-            initialize_function(self.m, 1 / (self.vp * self.vp), self.nbpml)
+            if vp.shape == self.vp.shape:
```

Comment:
I could use a comment here

Reply:
Well this mainly check if the new vp is padded or not and do the corresponding assignment